pandas/core/frame.py

```diff
@@ -1711,8 +1709,7 @@ def to_records(self, index=True, convert_datetime64=None,
             # string naming a type.
             if dtype_mapping is None:
                 formats.append(v.dtype)
-            elif isinstance(dtype_mapping, (type, np.dtype,
-                                            compat.string_types)):
+            elif isinstance(dtype_mapping, (type, np.dtype, str)):
```

Comment:
Just thinking out loud but wondering if we should make something to fuse type, np.dtype and str. Would certainly help out in some of the typing things we have going on